### PR TITLE
Project Pivot

### PR DESCRIPTION
**Pull request by DeronDMV**
Ran into a number of obstacles producing a satisfactory project under my original project premise. In the past few weeks Ive become a freelance digital marketing strategist for a new ride share app and find more interest and utility in exploring ride data, specifically, looking to see if I can predict whether or not a user who uses a discount code is an existing user or new user and the probability that either new/existing user will use a discount code for a trip. i've already cleaned the data set a bit and done some exploration. I would like to have this be the focus of my project if it will be accepted knowing the date for formal project changes has already passed. Do let me know

### CODE REVIEW

**Comment by sinanuozdemir** [comment]
Deron this sounds super awesome and exciting! I would love to discuss this further with you tonight just to make sure that you can get as much as possible out of this final project. I understand that sometimes we run into obstacles, but we're going to make sure that we get over them!

File: deron/Project/2PointB_Trips_Data_Exploration.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+Created on Mon Feb 16 20:35:35 2015
+
+@author: deronhogans
+"""
+
+import pandas as pd
+import numpy as np
+trips = pd.DataFrame.from_csv('2PointB_January_Trips.csv', sep= ",", header=0, index_col=0)
+
+trips['Status'] = trips.Status.map({'Meter Off': 1, 'Unable to Auth': 0, 'Cancelled':-1})
```
Comment (sinanuozdemir) [inline deron/Project/2PointB_Trips_Data_Exploration.py]: nice use of map!

File: deron/Project/2PointB_Trips_Data_Exploration.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+trips.Trip_Adjusted = trips.Trip_Adjusted.map({'Yes': 1, 'No': 0})
+
+trips.Existing_Passenger = trips.Existing_Passenger.map({'Yes': 1, 'No': 0})
+
+trips.dropna()
```
Comment (sinanuozdemir) [inline deron/Project/2PointB_Trips_Data_Exploration.py]: Remember to make sure you know that you're dropping all rows with ANY null values in them :)

**Comment by sinanuozdemir** [comment]
Let's also discuss the ideas of generating these probabilities. I smell a bayes theorem question :)

**Comment by DeronDMV** [comment]
Did a bit more research on Naive Bayes and I wonder, since Naive Bayes
functions best as a classifier, how would I best transform my question to
fit that function?

Currently my main question is:

*Can I use predict whether or not a promo code will be used for a ride and
how much based on Gross Trip Cost, Trip Adjustment and Existing Customer?*

I feel as though I would be classifying predictions as "Promo Code Used"
and "Promo Code Not Used", but what about predicting the the applied
discount?




On Wed, Feb 18, 2015 at 12:19 PM, Sinan Ozdemir <notifications@github.com>
wrote:

> Let's also discuss the ideas of generating these probabilities. I smell a
> bayes theorem question :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/justmarkham/DAT4-students/pull/86#issuecomment-74905590>
> .
>



-- 
Deron Hogans
C/O 2012
Communications, Culture, and Technology
Georgetown University
(202) 412 6709

**Comment by sinanuozdemir** [comment]
Good question! You're right that "Promo code used" and "'Promo code not used" is an option. If you would like to go the classification route, you could bucket prices as a category. It would look something like:

bucket A: discount < 5
bucket B:  5 <= discount <= 10
so on and so forth.

If you had more than two buckets, you can do a "one vs all approach" with naive bayes and calculate the probability for each bucket. I hope this makes sense. There is a great example of sklearn's MultinomialNB in the 13_naive_bayes.py file in the code section (line 113).

Hope this helps!

**Comment by DeronDMV** [comment]
Leave the jumbo slice to the beginners.
We know the District. We know food, and we know pizza. Join us for #ThatPizzaThing, one of those social jams with pizza curated by District Men. Enjoy Red Rocks' artisan food and drinks while you:

Dating advice from our resident Wing Woman
Field financial advice from Manage Your Damn Money
And a Dream for Purpose pop-up shop

Got questions? E-mail Deron, our editor-in-chief at DTHogans@DistrictMen202.Com
See you there!Share this event on Facebook
                    and Twitter
                    We hope you can make it!Cheers,District Men

------------------------------
Event Summary:
------------------------------

Event: #ThatPizzaThing
Date: Wednesday, October 7, 2015 from 5:00 PM to 10:00 PM (EDT)
Location: &lt;b&gt;1348 H St NE&lt;/b&gt;&lt;br /&gt;1348 H Street Northeast&lt;br /&gt;Washington, DC 20002&lt;br /&gt;

------------------------------
Event Details:
------------------------------

Leave the jumbo slice to the beginners.
We know the District. We know food, and we know pizza. Join us for #ThatPizzaThing, one of those social jams with pizza curated by District Men. Enjoy Red Rocks' artisan food and drinks while you:

Dating advice from our resident Wing Woman
Field financial advice from Manage Your Damn Money
And a Dream for Purpose pop-up shop

Got questions? E-mail Deron, our editor-in-chief at DTHogans@DistrictMen202.Com
See you there!

------------------------------
Hosted By:
------------------------------
District Men
Man is experiencing a shift in cultural awareness, and the District is too. The Nation’s Capital is no longer the sleepy, Southern city it used to be. Young professionals are making careers out of contributing innovative ideas to civilian markets, and seasoned executives are finding value in the District.
We are here to provide the lifestyle insight men need to succeed in the new D.C. The best bars? We drink there. The best tailors? They’re sewing up some new suits for us as we speak. We have a wealth of knowledge, so read up and get rich quick.
Contact us via e-mail at DistrictMen202@Gmail.Com

------------------------------
Register Online:
------------------------------

More information and online registration are available here:
https://www.eventbrite.com/e/thatpizzathing-tickets-18751547383?ref=enivtefor001&invite=ODUxODY2MC9yZXBseSswMDk3NTFhMzRkMTc0YmNjYzU2YWNhYmY2Y2FmNmU4MGQ3Njk2ODc3Njc3MTg1MWU5MmNmMDAwMDAwMDExMGZjOGI5NjkyYTE2OWNlMDM3MzNiNzlAcmVwbHkuZ2l0aHViLmNvbS8w

----------------------------------------------------------------------

Collect event fees online with Eventbrite
http://www.eventbrite.com

**Pull request merged**